google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardLibraryPanel.java

```diff
@@ -42,51 +43,77 @@
   private JPanel mainPanel;
   private JPanel libraryPanel;
 
+  private JCheckBox servletApiCheckbox;
+  private JCheckBox jstlCheckbox;
+  private JCheckBox appEngineApiCheckBox;
+  private JCheckBox endpointsCheckBox;
+  private JCheckBox objectifyCheckBox;
+
+  private List<JCheckBox> libraries = ImmutableList.of(servletApiCheckbox, jstlCheckbox,
+      appEngineApiCheckBox, endpointsCheckBox, objectifyCheckBox);
+
   private boolean enabled = true;
 
   public AppEngineStandardLibraryPanel(boolean enabled) {
     this.enabled = enabled;
+    libraryPanel.setVisible(enabled);
+
+    // The Servlet API is provided by the AE standard runtime. So we are enabling it by
+    // default.
+    servletApiCheckbox.setSelected(true);
+
+    // Objectify and Endpoints are dependencies of the App Engine API.
+    objectifyCheckBox.setSelected(false);
+    endpointsCheckBox.setSelected(false);
+    appEngineApiCheckBox.addItemListener(new AppEngineApiListener());
+    objectifyCheckBox.addItemListener(new AppEngineApiDependencyListener());
+    endpointsCheckBox.addItemListener(new AppEngineApiDependencyListener());
   }
 
   public Set<AppEngineStandardMavenLibrary> getSelectedLibraries() {
     return Sets.newHashSet(Collections2.filter(
```

Comment:
not blocking, but just looking at all this transformation code makes me think an immutable bimap would have been a simpler way to handle this.